pypykatz/commons/winapi/local/function_defs/live_reader_ctypes.py

```diff
@@ -49,41 +49,41 @@ class OSVersionInfo(ctypes.Structure):
 	PROCESS_ALL_ACCESS = STANDARD_RIGHTS_REQUIRED | SYNCHRONIZE | 0xFFFF
 else:
 	PROCESS_ALL_ACCESS = STANDARD_RIGHTS_REQUIRED | SYNCHRONIZE | 0xFFF
-	
+
 PROCESS_QUERY_INFORMATION = 0x0400
 PROCESS_VM_READ = 0x0010
 
-	
+
 #https://msdn.microsoft.com/en-us/library/windows/desktop/ms683217(v=vs.85).aspx
 def enum_process_names():
 	pid_to_name = {}
-	
+
 	for pid in EnumProcesses():
 		if pid == 0:
 			continue
 		pid_to_name[pid] = 'Not found'
 		try:
 			process_handle = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, False, pid)
+			pid_to_name[pid] = QueryFullProcessImageNameW(process_handle)
```

Comment:
Now, if `QueryFullProcessImageNameW(process_handle)` fails, the whole pypykatz won't. Chances are that the process name we need is already in the `pid_to_name` list.